File: 백준/Silver/10815. 숫자 카드/숫자 카드.cc
#include <iostream>
#include <algorithm>

using namespace std;

int search(int A[], int find ,int n) {
    int start=0;
    int end=n-1;
    int mid=0;

    while(start<=end) {
        mid = (start + end)/ 2;

        if(A[mid] == find) return 1;
        else if(A[mid] < find) start = mid+1;
        else if(A[mid] > find) end = mid-1;
    }
    return 0;
}
int main() {

    std::ios::sync_with_stdio(0);
    std::cin.tie(0);

    int n; cin >> n;

    int A[n]={};
    for(int i=0; i<n; i++) {
        cin >> A[i];
    }
    sort(A, A+n);

    int m; cin>>m;
    int B[m]={};
    for(int i=0; i< m; i++) {
        cin >> B[i];
    }


    for(int i=0 ; i< m; i++){
        cout << search(A, B[i],n) << " ";
    }

    return 0;
}

File: 백준/Silver/10815. 숫자 카드/숫자 카드.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

class Main {

    static int[] n;
    public static void main(String args[]) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int  N = Integer.parseInt(br.readLine());

        n = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());

        for(int i=0; i< N; i++){
            n[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(n);

        int M = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for(int i=0; i<M; i++){
            int x = Integer.parseInt(st.nextToken()) ;

            int answer = find(x);
            System.out.print(answer + " ");
        }

    }

    public static int find(int target){
        int start =0;
        int end = n.length-1;

        while(start<=end){
            int mid = (start + end)/2;

            if(n[mid] < target){
                start = mid+1;
            }else if(n[mid] > target){
                end = mid-1;
            }else if(n[mid] == target){
                return 1;
            }
        }
        return 0;
    }

}